node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -319,29 +320,62 @@ func (atp *apiTransactionProcessor) extractRequestedTxInfo(wrappedTx *txcache.Wr
 	}
 
 	if requestedFieldsHandler.HasSender {
-		tx.TxFields[senderField], _ = atp.addressPubKeyConverter.Encode(wrappedTx.Tx.GetSndAddr())
+		tx.TxFields[senderField] = atp.addressPubKeyConverter.SilentEncode(wrappedTx.Tx.GetSndAddr(), log)
```

Comment:
we can refactor the completion of the fields by the following way:

`requestedFieldHandler` will only expose a function: `IsFieldSet(fieldName) boolean`
and the logic of adding fields to the map can be something like this:

```
var fieldGetters = map[string]string{
	"receiver":     func() string { return atp.addressPubKeyConverter.SilentEncode(wrappedTx.Tx.GetRcvAddr(), log) },
	"gasLimit":     func() string { return wrappedTx.Tx.GetGasLimit() },
	"gasPrice":     func() string { return wrappedTx.Tx.GetGasPrice() },
	"rcvUsername": func() string { return wrappedTx.Tx.GetRcvUserName() },
	"data":         func() string { return wrappedTx.Tx.GetData() },
	"value":        func() string { return getTxValue(wrappedTx) },
        .......
}
...
if requestedFieldsHandler.IsFieldSet(wildCard) {
    // Set all fields
    for fieldName, getter := range fieldGetters {
        tx.TxFields[fieldName] = getter()
    }
} else {
    // Iterate over the map and execute the functions for the fields that are set in the request
    for fieldName, getter := range fieldGetters {
        if requestedFieldsHandler.IsFieldSet(fieldName) {
            tx.TxFields[fieldName] = getter()
        }
    }
}
```

optional, anyway.